File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.js
function solution(s){
    
    
    const ps = [...s.matchAll(/p/ig)];
    const ys = [...s.matchAll(/y/ig)];
    
    // p와 y 둘 다 없는 경우 
    if(ps === null && ys === null) return true;
    
    // p와 y의 개수가 같은 경우
    if(ps.length === ys.length){
        return true;
    }
    
    return false;
    
}

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.py
def solution(s):
    numOfp = 0
    numOfy = 0 
    
    for char in s:
        if char == 'p' or char == "P":
            numOfp += 1
        elif char == 'y' or char == "Y":
            numOfy += 1
            
    return numOfp == numOfy